src/modules/menu/components/advancedTools/ImportFile/utils.js

```diff
@@ -39,31 +41,57 @@ export function isGpx(fileContent) {
  * @param {OBject} store Vuex store
  * @param {ArrayBuffer} content Content of the file
  * @param {string} source Source of the file (either URL or file path)
+ * @param {File | null} [originalFile]
  * @returns {ExternalLayer} External layer object
  */
-export async function handleFileContent(store, content, source) {
+export async function handleFileContent(store, content, source, originalFile = null) {
     let layer = null
-    let textContent
+    let parsedContent
     let linkFiles
     if (isZipContent(content)) {
         log.debug(`File content is a zipfile, assume it is a KMZ archive`)
         const kmz = await unzipKmz(content, source)
-        textContent = kmz.kml
+        parsedContent = kmz.kml
         linkFiles = kmz.files
+    } else if (source.endsWith('.tif')) {
+        log.debug(`File content might be a COGTIFF, attempting a parse as such`)
+        try {
+            parsedContent = await fromArrayBuffer(content)
+        } catch (err) {
+            log.debug('parsing as COGTIFF failed, moving on with other formats', err)
```

Comment:
Here it should be an error and we should throw an error in order to provide a user feedback that the tif file is invalid